syne_tune/optimizer/schedulers/transfer_learning/zero_shot.py

```diff
@@ -0,0 +1,151 @@
+            if self._mode == 'max':
+                avg_scores = avg_scores.max(axis=1)[idx]
+            else:
+                avg_scores = avg_scores.min(axis=1)[idx]
+            scores.append(avg_scores)
+        if not use_surrogates:
+            if len(transfer_learning_evaluations) > 1:
+                logger.warning(warning_message + 'If this is not the case, this searcher fails without a warning.')
+            if not sort_transfer_learning_evaluations:
+                hyperparameters = hyperparameters.copy()
+        hyperparameters.reset_index(drop=True, inplace=True)
+        self._hyperparameters = hyperparameters
+        sign = 1 if self._mode == 'min' else -1
+        self._scores = sign * pd.DataFrame(scores)
+        self._ranks = self._update_ranks()
+
+    def _create_surrogate_transfer_learning_evaluations(
+            self, config_space: Dict, transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            metric: str) -> Dict[str, TransferLearningTaskEvaluations]:
+        """
+        Creates transfer_learning_evaluations where each configuration is evaluated on each task using surrogate models.
+        """
+        surrogate_transfer_learning_evaluations = dict()
+        for task_name, task_data in transfer_learning_evaluations.items():
+            estimator = BlackboxSurrogate.make_model_pipeline(
+                configuration_space=config_space,
+                fidelity_space={},
+                model=xgboost.XGBRegressor(),
+            )
+            X_train = task_data.hyperparameters
+            y_train = task_data.objective_values(metric).mean(axis=1)
+            if self._mode == 'max':
+                y_train = y_train.max(axis=1)
+            else:
+                y_train = y_train.min(axis=1)
+            estimator.fit(X_train, y_train)
+
+            num_candidates = 10000 if len(config_space) >= 6 else 5 ** len(config_space)
+            hyperparameters_new = pd.DataFrame([self._sample_random_config(config_space)
+                                                for _ in range(num_candidates)])
+            objectives_evaluations_new = estimator.predict(hyperparameters_new).reshape(-1, 1, 1, 1)
+            surrogate_transfer_learning_evaluations[task_name] = TransferLearningTaskEvaluations(
+                configuration_space=config_space, hyperparameters=hyperparameters_new, objectives_names=[metric],
+                objectives_evaluations=objectives_evaluations_new
+            )
+        return surrogate_transfer_learning_evaluations
+
+    def get_config(self, **kwargs) -> Optional[Dict]:
+        if self._ranks.shape[1] == 0:
+            return None
+        best_idx = self._ranks.mean(axis=0).idxmin()
+        self._ranks.clip(upper=self._ranks[best_idx], axis=0, inplace=True)
+        self._scores.drop(columns=best_idx, inplace=True)
+        best_config = self._hyperparameters.loc[best_idx]
```

Comment:
Those lines would probably deserve a comment given that it is the hearth of the method but understanding what is happening is not straightforward just given the lines in pandas.